### PR TITLE
Using a list-if expression instead of a ternary operator

### PR DESCRIPTION
**Pull request by domesticmouse**
An alternate to https://github.com/flutter/codelabs/pull/1455

## Pre-launch Checklist

- [x] I read the [Effective Dart: Style] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-devrel channel on [Discord].

[Effective Dart: Style]: https://dart.dev/guides/language/effective-dart/style
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: cookbook/lib/examples/gradient_bubbles.dart [hunk 1]
```diff
@@ -86,12 +86,14 @@ class MessageBubble extends StatelessWidget {
...
-                    ? const Color(0xFF3A364B)
-                    : const Color(0xFF491CCB),
+                if (message.isMine) ...[
+                  const Color(0xFF6C7689),
+                  const Color(0xFF3A364B),
+                ],
+                if (!message.isMine) ...[
+                  const Color(0xFF19B7FF),
+                  const Color(0xFF491CCB),
+                ],
```
Comment (parlough) [inline cookbook/lib/examples/gradient_bubbles.dart:96]: Can't you use an `else` here? Or are you keeping them separate to illustrate something more clearly?

```suggestion
                if (message.isMine) ...[
                  const Color(0xFF6C7689),
                  const Color(0xFF3A364B),
                ] else ...[
                  const Color(0xFF19B7FF),
                  const Color(0xFF491CCB),
                ],
```

Comment (domesticmouse) [reply cookbook/lib/examples/gradient_bubbles.dart]: For some reason I thought `else` wasn't available here. Today, I learned, and all that =)

Comment (parlough) [reply cookbook/lib/examples/gradient_bubbles.dart]: We still don't document collection control flow expressions well :( so a lot of people may not know.

https://github.com/dart-lang/site-www/issues/2093

Now I see the language tour doesn't even have an example with an `else`.

File: cookbook/lib/examples/gradient_bubbles.dart [hunk 2]
```diff
@@ -86,12 +86,13 @@ class MessageBubble extends StatelessWidget {
...
-                message.isMine
-                    ? const Color(0xFF3A364B)
-                    : const Color(0xFF491CCB),
+                if (message.isMine) ...[
+                  const Color(0xFF6C7689),
+                  const Color(0xFF3A364B),
+                ] else ...[
+                  const Color(0xFF19B7FF),
+                  const Color(0xFF491CCB),
+                ],
```
Comment (parlough) [inline cookbook/lib/examples/gradient_bubbles.dart:95]: Consider pulling the `const`s out:

```suggestion
                if (message.isMine) ...const [
                  Color(0xFF6C7689),
                  Color(0xFF3A364B),
                ] else ...const [
                  Color(0xFF19B7FF),
                  Color(0xFF491CCB),
                ],
```

**Review by parlough — APPROVED** [review]
Avoiding ternaries always seems good to me :)

One consideration:

**Pull request merged**